### PR TITLE
test: add unit tests for trunk based development.

### PR DESCRIPTION
**Pull request by ganesh-nithin**
# Why this change is required

This PR aims to add unit tests for trunk based development.

# Describe your changes

- added unit test cases.

# How this change has been tested

Tested with jest test runner.

# Reference

image

# Checklist

- [x] I have performed a self review of my code and intend to submit as such
- [ ] I have added necessary explanations and inline comments for review
- [ ] I have considered updating necessary README or other documentations
- [x] I have added/modified necessary automated tests
- [ ] This includes a breaking changes and needs to be listed in release notes

### CODE REVIEW

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+import { ServerConfiguration } from '##/configs/server.config.js';
+
+const {
+  organization: ORGANIZATION,
+  projectName: PROJECT,
+  repositoryId: REPOSITORY_ID,
+} = ServerConfiguration.versionControl;
+
+describe('TrunkBasedDevelopment~getBranchMetrics - method to transform branches into following and not following branch naming standard', () => {
```
Comment (sathish-thangavel) [inline server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js:13]: `method to get branches following and not following naming standard`

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    ];
+
+    const result = TrunkBasedDevelopment.getBranchMetrics({ value: branches, count: branches.length });
+
+    expect(result).toEqual({
+      branchesURL: `https://dev.azure.com/${ORGANIZATION}/${PROJECT}/_git/${REPOSITORY_ID}/branches?a=all&_a=all`,
+      totalNumberOfBranches: 4,
+      percentageOfBranchesFollowingStandard: '50.00%',
+      branchesFollowingNamingStandard: {
+        count: 2,
```
Comment (sathish-thangavel) [inline server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js:29]: Can we move this to constant

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+  it('should handle no branches', () => {
+    const branches = [];
+
+    const result = TrunkBasedDevelopment.getBranchMetrics({ value: branches, count: branches.length });
+
+    expect(result).toEqual({
+      branchesURL: `https://dev.azure.com/${ORGANIZATION}/${PROJECT}/_git/${REPOSITORY_ID}/branches?a=all&_a=all`,
+      totalNumberOfBranches: 0,
+      percentageOfBranchesFollowingStandard: 'NaN%',
```
Comment (sathish-thangavel) [inline server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js:69]: Should be fixed and should be `0%`

**Review by sathish-thangavel — CHANGES_REQUESTED** [review]

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
+import { describe, it, expect } from '@jest/globals';
+
+import { TrunkBasedDevelopment } from './trunk-based-development.js';
+
```
[the remark this answers is not in the document]
Comment (ganesh-nithin) [reply server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js:4]: No, Sathish. I have added them.

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.js [hunk 5]
```diff
@@ -27,8 +27,11 @@ export class TrunkBasedDevelopment {
       { branchesFollowingNamingStandard: [], branchesNotFollowingNamingStandard: [] }
     );
 
-    const percentageOfBranchesFollowingStandard =
-      ((branchesFollowingNamingStandard.length / totalNumberOfBranches) * MAX_PERCENTAGE).toFixed(2) + PERCENTAGE;
+    const zeroPercentage = 0 + PERCENTAGE;
+
+    const percentageOfBranchesFollowingStandard = !totalNumberOfBranches
+      ? zeroPercentage
```
Comment (sathish-thangavel) [inline server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.js:33]: ``0${PERCENTAGE}``

File: server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { describe, it, expect } from '@jest/globals';
+
+import { TrunkBasedDevelopment } from './trunk-based-development.js';
+import {
+  AZURE_ACTIVE_BRANCHES_RESPONSE,
+  AZURE_ALL_BRANCHES_RESPONSE,
+  AZURE_COUNT_ZERO_RESPONSE,
```
Comment (sathish-thangavel) [inline server/src/use-cases/version-control/azure-devops/trunk-based-development/trunk-based-development.unit.test.js:7]: `import * as mock from './';`

**Review by sathish-thangavel — CHANGES_REQUESTED** [review]

**Review by sathish-thangavel — APPROVED** [review]

**Pull request merged**